ibm/resource_ibm_cse_instance.go

```diff
@@ -0,0 +1,388 @@
+				Optional:    true,
+			},
+			"multi_tenant": &schema.Schema{
+				Type:        schema.TypeInt,
+				Description: "The dedicated device will only run this CSE when the value is 1.",
+				Optional:    true,
+			},
+			"acl": &schema.Schema{
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "The acl setting for this CSE.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Set:         schema.HashString,
+			},
+			"url": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"service_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"static_addresses": &schema.Schema{
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCSEInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	cseClient, err := meta.(ClientSession).CseAPI()
+	if err != nil {
+		return err
+	}
+
+	seAPI := cseClient.ServiceEndpoints()
+
+	payload, err := genCreatePayload(d)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("resourceCSEInstanceCreate: payload=%v\n", payload)
+
+	srvId, err := seAPI.CreateServiceEndpoint(payload)
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(srvId)
+	d.Set("service_id", srvId)
+
+	return nil
+}
+
+func resourceCSEInstanceRead(d *schema.ResourceData, meta interface{}) error {
+	cseClient, err := meta.(ClientSession).CseAPI()
+	if err != nil {
+		return err
+	}
+
+	seAPI := cseClient.ServiceEndpoints()
+
+	srvId := d.Get("service_id").(string)
+
+	srvObj, err := seAPI.GetServiceEndpoint(srvId)
+
+	if err != nil {
+		return err
+	}
+
+	d.Set("url", srvObj.Service.URL)
+	addresses := []string{}
+	for _, se := range srvObj.Endpoints {
+		addresses = append(addresses, se.StaticAddress)
+	}
+
+	d.Set("static_addresses", addresses)
+
+	return nil
+}
+
+func resourceCSEInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+	payload, err := genUpdatePayload(d)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("resourceCSEInstanceUpdate: payload=%v", payload)
+
+	cseClient, err := meta.(ClientSession).CseAPI()
+	if err != nil {
+		return err
+	}
+
+	seAPI := cseClient.ServiceEndpoints()
+
+	srvId := d.Get("service_id").(string)
+
+	err = seAPI.UpdateServiceEndpoint(srvId, payload)
+
+	if err != nil {
+		return err
+	}
+
+	return resourceCSEInstanceRead(d, meta)
+}
+
+func resourceCSEInstanceDelete(d *schema.ResourceData, meta interface{}) error {
+	cseClient, err := meta.(ClientSession).CseAPI()
+	if err != nil {
+		return err
+	}
+
+	seAPI := cseClient.ServiceEndpoints()
+
+	srvId := d.Get("service_id").(string)
+
+	err = seAPI.DeleteServiceEndpoint(srvId)
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func resourceCSEInstanceExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	return true, nil
```

Comment:
Please add the logic to check whether instance exists via calling Get Method..some sample logic can be found in other resources.

Reply:
Will add it asap